gdk4/src/rectangle.rs

```diff
@@ -0,0 +1,167 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use cairo::RectangleInt;
+use glib::ffi::gconstpointer;
+use glib::translate::*;
+use std::convert::{AsRef, From};
+use std::mem;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct Rectangle {
+    pub x: i32,
+    pub y: i32,
+    pub width: i32,
+    pub height: i32,
+}
+
+impl Rectangle {
+    pub fn intersect(&self, other: &Rectangle) -> Option<Rectangle> {
```

Comment:
You forgot the `contains_point()` function

Reply:
Added contains_point and equal